app/javascript/components/public_health/query/AdvancedFilter.js

```diff
@@ -757,30 +777,124 @@ class AdvancedFilter extends React.Component {
+              )}
+              {this.state.activeFilter && (
+                <Button id="advanced-filter-update" variant="primary" onClick={this.update} className="mr-1">
+                  <i className="fas fa-marker"></i>
+                  <span className="ml-1">Update</span>
+                </Button>
+              )}
+              {this.state.activeFilter && (
+                <Button id="advanced-filter-delete" variant="danger" onClick={this.delete} disabled={!this.state.activeFilter}>
+                  <i className="fas fa-trash"></i>
+                  <span className="ml-1">Delete</span>
+                </Button>
+              )}
+              <div className="float-right">
+                <Button id="advanced-filter-reset" variant="danger" onClick={this.reset}>
+                  Reset
+                </Button>
+                <Button id="advanced-filter-apply" variant="primary" className="ml-2" onClick={this.apply}>
+                  Apply
+                </Button>
+              </div>
+            </Col>
+          </Row>
+          <Row>
+            <Col className="pb-3 pt-1">
+              <div className="g-border-bottom-2"></div>
+            </Col>
+          </Row>
+          {this.state.activeFilterOptions?.map((statement, index) => {
+            return this.renderStatement(
+              statement.filterOption,
+              statement.value,
+              index,
+              this.state.activeFilterOptions?.length,
+              statement.numberOption,
+              statement.dateOption,
+              statement.relativeOption,
+              statement.additionalFilterOption
+            );
+          })}
+          <Row className="pt-2 pb-1">
+            <Col>
+              <Button
+                id="add-filter-row"
+                variant="primary"
+                disabled={this.state.activeFilterOptions?.length > 4}
+                onClick={() => this.add()}
+                aria-label="Add Advanced Filter Option">
+                <i className="fas fa-plus"></i>
+              </Button>
+            </Col>
+          </Row>
+        </Modal.Body>
+        <Modal.Footer className="justify-unset">
+          <p className="lead mr-auto">
+            Filter will be applied to the line lists in the <u>{this.props.workflow}</u> workflow until reset.
+          </p>
+          <Button id="advanced-filter-cancel" variant="secondary btn-square" onClick={this.cancel}>
+            Cancel
+          </Button>
+        </Modal.Footer>
+      </Modal>
+    );
+  };
+
   // Render a single line "statement"
   renderStatement = (filterOption, value, index, total, numberOption, dateOption, relativeOption, additionalFilterOption) => {
     return (
       <React.Fragment key={'rowkey-filter-p' + index}>
         {index > 0 && index < total && (
-          <Row key={'rowkey-filter-and' + index} className="pb-2 pt-2">
+          <Row key={'rowkey-filter-and' + index} className="and-row pb-2 pt-2">
```

Comment:
out of curiosity, is this `and-row` class just used by the tests to reference this DOM element?

Reply:
yes it is